OpenSim/Common/Property.h

```diff
@@ -694,6 +696,101 @@ Property<T>::getTypeName() const {
 // Hide SimpleProperty and ObjectProperty from Doxygen; users don't need
 // to know about these.
 /** @cond **/
+
+//==============================================================================
+//                  HELPERS FOR WRITING PROPERTY VALUES 
+//==============================================================================
+// This section is hidden from SWIG because of compiling issues (i.e., SWIG
+// thought these functions were part of Property in some cases, rather than
+// free functions)
+#ifndef SWIG
+inline void convertTransformToVec6(SimTK::Array_<SimTK::Vec6>& rotTrans,
+    const SimTK::Transform& transform)
+{
+    SimTK::Vec6 X6;
+    SimTK::Vec3& angles = X6.updSubVec<3>(0);
+    SimTK::Vec3& pos = X6.updSubVec<3>(3);
+    angles = transform.R().convertRotationToBodyFixedXYZ();
+    pos = transform.p();
+    rotTrans.push_back(X6);
+}
+
+template <class T> inline void
+writeSimplePropertyToStreamForDisplay(std::ostream& o,
+    T& v, const int precision)
+{
+    SimTK::writeUnformatted(o, v);
+}
+
+inline void
+writeSimplePropertyToStreamForDisplay(std::ostream& o,
+    const double v, const int precision)
```

Comment:
When I looked into `SimTK::writeUnformatted()` a bit more, it seemed to be that it's actually a bunch of overloading with some templates to reduce code duplication. I chose to limit the templatization here rather than do all of the full blown stuff `writeUnformatted` does since OpenSim doesn't need all of those types. Might be more future-proof though if more types are covered?

Reply:
Oh okay; this is fine. I don't think we need to cover all types.